src/drivers/intel/dmic/dmic.c

```diff
@@ -369,16 +372,6 @@ static void dmic_start(struct dai *dai)
 	dmic->state = COMP_STATE_ACTIVE;
 	k_spin_unlock(&dai->lock, key);
 
-	/* Currently there's no DMIC HW internal mutings and wait times
-	 * applied into this start sequence. It can be implemented here if
-	 * start of audio capture would contain clicks and/or noise and it
-	 * is not suppressed by gain ramp somewhere in the capture pipe.
-	 */
-
-	schedule_task(&dmic->dmicwork, DMIC_UNMUTE_RAMP_US,
```

Comment:
Variable dmic->gain_coef is computed from DMIC_UNMUTE_RAMP_US. Would you also update the equation in dmic.h and dmic.c for actual rate of this callback? This works only for 1 ms now.

Reply:
Does it work for event driven scheduling?

Reply:
@singalsu Scheduling can be timer- or DMA-based with SOF. DMA-based scheduling is used by non-Intel platforms, by older Intel platforms like BYT, and in some rare cases, like with codec clock sourcing. So, in CI, I think DMA scheduling is only used with pre-cAVS Intel platforms with with non-Intel platforms. Do any of them use DMIC?

Concerning non-default scheduling period, when the timer-driven pipeline has a non-1ms period. This is less clear to me. You're saying that if, say, a pipeline would be scheduled every 10ms, that would be too coarse for DMIC ramping? But is this ever done with DMIC pipelines? And if for some reason someone does decide to have such a scheduling period, maybe a more coarse ramping is acceptable to them?

Reply:
Not sure I follow, but you can test that you see a ramp that has the length that is described in topology in DAI DMIC parameters (IPC3) or get the correct hard coded ramp time (IPC4). Change pipeline scheduling to e.g. 2 ms and check that you still get correct ramp time. We could address the zipper noise with long period pipelines later. It's a subjective issue. If the pipeline is for machine hearing it doesn't matter.